File: nodejs/checkio/src/home/first_word.ts
import assert from "assert";

function firstWord(text: string): string {
  const words = text.split(/[\s,.]+/);
  let first = "";
  for (const str of words) {
    if (/\w/.test(str)) {
      first = str;
      break;
    }
  }
  return first;
}

console.log("Example:");
console.log(firstWord("Hello world"));

// These "asserts" are used for self-checking
assert.strictEqual(firstWord("Hello world"), "Hello");
assert.strictEqual(firstWord(" a word "), "a");
assert.strictEqual(firstWord("don't touch it"), "don't");
assert.strictEqual(firstWord("greetings, friends"), "greetings");
assert.strictEqual(firstWord("... and so on ..."), "and");
assert.strictEqual(firstWord("hi"), "hi");

console.log("Coding complete? Click 'Check Solution' to earn rewards!");

File: python/checkio/home/first-word.py
import re


def first_word(text: str) -> str:
    for s in text.strip().split():
        if s[0].isalpha():
            return re.split(r"[,\.]", s)[0]
    return ""


print("Example:")
print(first_word("Hello world"))

# These "asserts" are used for self-checking
assert first_word("Hello world") == "Hello"
assert first_word(" a word ") == "a"
assert first_word("don't touch it") == "don't"
assert first_word("greetings, friends") == "greetings"
assert first_word("... and so on ...") == "and"
assert first_word("hi") == "hi"

print("The mission is done! Click 'Check Solution' to earn rewards!")
